### DISCUSSION TITLE
is grafana need to mount data directory to PVC?

### DISCUSSION THREAD

**Discussion by hillbun** [prometheus-operator/kube-prometheus#969] [Q&A]
if yes, which is  grafana data directory?

**Comment by paulfantom** [comment]
No, stateless grafana provided by this project by default doesn't need PVC as all configuration is stored either as a ConfigMap or Secret.

**Comment by hillbun** [comment]
@paulfantom 

but I change admin password of grafana, when pod recreated, password change to default values.

so does it need persistence?

**Reply by paulfantom** [reply]
That's the immutability aspect of the default grafana installation provided by kube-prometheus. You have two options:
- change password (and any other setting) in ConfigMap
- add PVC to grafana and make it stateful and mutable

**Reply by kaushiklala** [reply]
> That's the immutability aspect of the default grafana installation provided by kube-prometheus. You have two options:
> 
> * change password (and any other setting) in ConfigMap
> * add PVC to grafana and make it stateful and mutable

@paulfantom Is there any jsonnet example on how to add a pvc for grafana ?

**Comment by hillbun** [comment]
@paulfantom 

I can not find ConfigMap setting for grafana logins.
please advice.
